src/Event/Productions/ProductionCommandHandler.php

```diff
@@ -104,4 +121,12 @@ private function productionsWereMerged(
         }
         $this->similaritiesClient->excludeTemporarily($eventPairs);
     }
+
+    private function assertEventExists(string $eventId)
+    {
+        $event = $this->eventRepository->get($eventId);
```

Comment:
I think this can throw an exception? At least the interface also show a `DocumentGoneException` exception.

Reply:
The concrete implementation `CacheDocumentRepository` checks for `GONE` returned from the cache and in that case throws a `DocumentGoneException`.